keps/sig-apps/0753-sidecarcontainers.md

```diff
@@ -248,11 +458,833 @@ Package `kuberuntime`
+
+#### Sidecar containers won't be available during initContainers phase
+
+The current proposal adds sidecar containers that will be started after all
+initContainer executed. In other words, it just gives an order to containers on
+the `containers` array.
+
+For most users we talked about (some big companies and linkerd, will try to
+contact istio soon) this doesn't seem like a problem. But wanted to make this
+visible, just in case.
+
+Furthermore, service meshes don't provide easy ways to use them with
+initContainers (probably due to Kubernetes limitations). This KEP won't change
+that.
+
+Another thing to take into account is that Istio has an alternative to the
+initContainer approach. Istio [has an option][istio-cni-opt] to integrate with
+CNI and inject the blackhole from there instead of using the initContainer. In
+that case, it will do (just c&p from the link, in case it breaks in the
+future):
+
+> By default Istio injects an initContainer, istio-init, in pods deployed in the mesh. The istio-init container sets up the pod network traffic redirection to/from the Istio sidecar proxy. This requires the user or service-account deploying pods to the mesh to have sufficient Kubernetes RBAC permissions to deploy containers with the NET_ADMIN and NET_RAW capabilities. Requiring Istio users to have elevated Kubernetes RBAC permissions is problematic for some organizations’ security compliance
+> ...
+> The Istio CNI plugin performs the Istio mesh pod traffic redirection in the Kubernetes pod lifecycle’s network setup phase, thereby removing the requirement for the NET_ADMIN and NET_RAW capabilities for users deploying pods into the Istio mesh. The Istio CNI plugin replaces the functionality provided by the istio-init container.
+
+In other words, when using the CNI plugin it seems that InitContainer don't use
+the service mesh either. Rodrigo will double check this, just in case.
+
+[linkerd-last-container]: https://github.com/linkerd/linkerd2/issues/4758#issuecomment-658457737
+[istio-cni-opt]: https://istio.io/latest/docs/setup/additional-setup/cni/
+
+##### Suggestion
+
+Confirm with users that is okay to not have sidecars during the initContainers
+phase and they don't foresee any reason to add them in the near future.
+
+It seems likely that is not a problem and this is a win for them, as they can
+remove most of the hacks. However, it seems worth investigating if the CNI
+plugin is a viable alternative for most service mesh and, in that case, how much
+they will benefit from this sidecar KEP.
+
+It seems likely that they will benefit for two reason: (a) this might
+be simpler or model better what service mesh need during startup, (b) they still
+need to solve the problem on shutdown, where the service mesh needs to drain
+connections first and be running until others terminate. These are needed for
+graceful shutdown and allowing other containers to use the network on shutdown,
+respectively.
```

Comment:
I got a little confused by this section. The CNI plugin is only being used just for setting up the iptables forwarding. The main goal of this KEP (IMO), i.e. setting the proper startup and shutdown ordering among sidecars and regular containers, remains a critical need regardless of whether the forwarding rules are set through CNI or through an initContainer.

Reply:
Yes, sorry, it was a misunderstanding from my side. As it was explained [in this comment][link], will update the text regarding this :)

[link]: https://github.com/kubernetes/enhancements/pull/1913#discussion_r462571359